p2p/p2p.go

```diff
@@ -76,6 +77,7 @@ func NewTCPNode(ctx context.Context, cfg Config, key *k1.PrivateKey, connGater C
 			return filterAdvertisedAddrs(externalAddrs, internalAddrs, filterPrivateAddrs)
 		}),
 		libp2p.Transport(tcp.NewTCPTransport, tcpOpts...),
+		libp2p.SwarmOpts(swarm.WithDialRanker(swarm.NoDelayDialRanker)),
```

Comment:
Afaict from the [libp2p docs](https://pkg.go.dev/github.com/libp2p/go-libp2p/p2p/net/swarm#DefaultDialRanker), this way we will make attempts at the same time to connect to the private/public libp2p node address and also to the relay address, without prioritising the former? Was it like that before? I suppose practically speaking the attempt for direct connection to the libp2p node will evaluate faster.

Reply:
This is essentially what was happening before libp2p 0.32.0, we just restored the original behavior.

Also, to be precise, we're setting the `AddrDelay` provider to be `swarm.NoDelayDialRanker`, which as the docs say:

>NoDelayDialRanker ranks addresses with no delay. This is useful for simultaneous connect requests.

Reply:
Gotcha, thanks!